packages/amplify_datastore_plugin_interface/lib/src/types/temporal/temporal_time.dart

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'temporal.dart';
+
+/// Represents a valid extended ISO-8601 Time string, with an optional timezone offset.
+/// hh:mm:ss.sss±hh:mm:ss
+/// hh:mm:ss.sss
+/// https://docs.aws.amazon.com/appsync/latest/devguide/scalars.html#appsync-defined-scalars
+class TemporalTime {
+  DateTime _dateTime;
+  int _nanoseconds; // DateTime only stores millisecond and microsecond
```

Comment:
It's unfortunate that a Dart `DateTime` can't store nanoseconds :unamused:

From what I can tell, it does look like you are correctly saving the milliseconds, microseconds, and nanoseconds within this member variable though.

One minor suggestion though - I think it might simplify some of the conversion logic if you store the milliseconds and microseconds as a part of `_dateTime`, and then just use `_nanoseconds` for just the nanoseconds components (a value 0...999).  Then, the `toDateTime` method can be simplified to a simple getter, which just returns `_dateTime`.

Reply:
Changed code based on your minor suggestion though the changes require more complexity in the fromString and format methods in the end.